mlrun/projects/pipelines.py

```diff
@@ -708,11 +707,22 @@ def run(
         secrets=None,
         artifact_path=None,
         namespace=None,
+        source=None,
     ) -> typing.Optional[_PipelineRunStatus]:
         workflow_name = name.split("-")[-1] if f"{project.name}-" in name else name
         runner_name = f"workflow-runner-{workflow_name}"
         run_id = None
 
+        save, project_source = (
+            (False, source) if source else (True, project.spec.source)
+        )
+
```

Comment:
this one liner requires an explanation why we wouldn't want to save  if source is provided

Reply:
also, it the `project_source` is the exact name? 🤔

Reply:
will add an explanation, what name u suggest? `url`?